spec/e2e/validation_association_spec.rb

```diff
@@ -0,0 +1,57 @@
+describe Neo4j::ActiveNode::Validations do
+  before(:each) do
+    stub_active_node_class('Comment')
+
+    stub_active_node_class('Post') do
+      has_many :out, :comments, type: :COMMENT
+
+      validates :comments, presence: true
+    end
+  end
+
+  context 'validating presence' do
+    it 'new object should not be valid without comments' do
+      expect(Post.new({})).not_to be_valid
+    end
+
+    it 'should not be valid without comments' do
+      expect(Post.create).not_to be_valid
+    end
+
+    it 'should be valid with comments' do
+      expect(Post.new(comments: [Comment.create])).to be_valid
+    end
+  end
+
+  #The below spec pass on active_record as is
+  context 'active_record behaviour' do
+    before :each do
+      @post = Post.create(comments: [Comment.create])
+    end
```

Comment:
We try to avoid the use of instance variables in our specs these days. Instead:

```ruby
let(:post) { Post.create(comments: [Comment.create]) }
```

And then you can just call `post` in your examples.